r/streaming_pipe_in.R

```diff
@@ -0,0 +1,21 @@
+# Script reads text from a named pipe and writes it to stdout, capitalized
+# Command-line arguments
+args <- commandArgs(trailingOnly = TRUE)
+
+# Named pipe path (input)
+pipe_in <- args[1]
+
+# Open named pipe for reading
+input <- file(pipe_in, "r")
+
+# Read each line from input pipe, convert to uppercase, and print to stdout
+while (TRUE) {
+  line <- readLines(input, n = 1)
+  if (length(line) == 0) {
+    break  # Exit loop when there's no more data to read
+  }
```

Comment:
```suggestion
while (length(line <- readLines(input, n = 1)) > 0) {
```

(a more succinct way to write it)